samcli/commands/package/package_context.py

```diff
@@ -0,0 +1,129 @@
+"""
+Logic for uploading to s3 based on supplied template file and s3 bucket
+"""
+
+# Copyright 2012-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import os
+import logging
+import sys
+
+import json
+
+from samcli.commands.package.artifact_exporter import Template
+from samcli.yamlhelper import yaml_dump
+from samcli.commands.package import exceptions
+from samcli.commands.package.s3_uploader import S3Uploader
+
+import boto3
+from botocore.config import Config
+
+LOG = logging.getLogger(__name__)
+
+
+class PackageCommandContext(object):
+
+    MSG_PACKAGED_TEMPLATE_WRITTEN = (
+        "Successfully packaged artifacts and wrote output template "
+        "to file {output_file_name}."
+        "\n"
+        "Execute the following command to deploy the packaged template"
+        "\n"
+        "sam deploy --template-file {output_file_path} "
+        "--stack-name <YOUR STACK NAME>"
+        "\n"
+    )
+
+    NAME = "package"
+
+    def __init__(
+        self,
+        template_file,
+        s3_bucket,
+        s3_prefix,
+        kms_key_id,
+        output_template_file,
+        use_json,
+        force_upload,
+        metadata,
+        region,
+        profile,
+    ):
+        self.template_file = template_file
+        self.s3_bucket = s3_bucket
+        self.s3_prefix = s3_prefix
+        self.kms_key_id = kms_key_id
+        self.output_template_file = output_template_file
+        self.use_json = use_json
+        self.force_upload = force_upload
+        self.metadata = metadata
+        self.region = region
+        self.profile = profile
+        self.s3_uploader = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def run(self):
+
+        session = boto3.Session(profile_name=self.profile if self.profile else None)
+        s3_client = session.client(
+            "s3", config=Config(signature_version="s3v4", region_name=self.region if self.region else None)
+        )
+
+        template_path = self.template_file
+        if not os.path.isfile(template_path):
```

Comment:
This check isn't needed if you expand the `click.Path()` or change it to `click.File()` in the `--template-file` option on the command. 

https://click.palletsprojects.com/en/7.x/api/#click.Path